src/components/editor.tsx

```diff
@@ -63,10 +63,11 @@ export const Editor: FC<EditorProperties> = ({
 		highlightedRangesExtension(highlightedRanges),
 	];
 
-	const debouncedOnChange = useCallback(
-		debounce((value: string) => {
-			onChange?.(value);
-		}, 400),
+	const debouncedOnChange = useMemo(
+		() =>
+			debounce((value: string) => {
+				onChange?.(value);
+			}, 400),
```

Comment:
```

const debouncedOnChange = useMemo(
		() =>
			debounce((value: string) => {
				onChange?.(value);
			}, 400),
		[onChange],
	);
	
```



How does `useMemo` help in this case? We are returning a function, not a value. `useCallback` is preferred when using a function. In this case, if we use `useMemo`, it wouldn't have any impact, right? Is my understanding correct, or am I missing something?

Reply:
Functionally, it is equivalent, see also https://react.dev/reference/react/useCallback#how-is-usecallback-related-to-usememo  
But for whatever reason, the react eslint plugin would warn when using `useCallback`:

> React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead. eslint([react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920))

That error vanishes with `useMemo`.